### ISSUE TITLE
Minting via Etherscan has different gas rate than dapp?

### ISSUE THREAD

**Issue by basememara** [hashlips-lab/nft-erc721-collection#172]
First off.. great work on this smart contract and dapp!! The code is so simple and extensible. Thank you for sharing and supporting the community!

I'm hoping someone can help with a concern I'm experiencing and not sure if it's me or what I'm missing. When I deployed the smart contract to mainnet, I tried to mint via Etherscan, but the gas fee for minting just one NFT was 1.5 ETH! I tried disconnecting, reconnecting, refresh and same.

However, when minting one through the dapp, it was the expected gas rate of like $15-20. When I went back to Etherscan to trying minting from there again, the gas estimate went back to normal like the dapp like $15-20.

Does anyone understand what happened here? Has anyone else experiencing this or any ideas?

**Comment by liarco** [comment]
Hi @basememara, thank you for your feedback.

The gas estimation can be handled on the DAPP side in a custom way. My implementation is simply relaying on the web3 provider completely (so it's Infura, when using MetaMask), but another DAPP like Etherscan might be using another strategy to estimate the gas fee depending on the network congestion (even if you are using the same wallet through MetaMask).

Long story short, you can always expect different fees from different DAPPs, but unfortunately there is nothing we can do about how Etherscan handles these things.
